rsocket/framing/FramedReader.cpp

```diff
@@ -12,214 +12,206 @@ namespace rsocket {
+    // After this call the instance can be destroyed!
     subscriber->onError(std::move(ex));
   }
 }
 
-void FramedReader::request(int64_t n) noexcept {
+void FramedReader::request(int64_t n) {
   allowance_.release(n);
   parseFrames();
 }
 
-void FramedReader::cancel() noexcept {
+void FramedReader::cancel() {
   allowance_.drain();
-  frames_ = nullptr;
+  inner_ = nullptr;
 }
 
 void FramedReader::setInput(
-    yarpl::Reference<DuplexConnection::Subscriber> frames) {
-  CHECK(!frames_)
-      << "FrameReader should be closed before setting another input.";
-  frames_ = std::move(frames);
-  frames_->onSubscribe(this->ref_from_this(this));
+    yarpl::Reference<DuplexConnection::Subscriber> inner) {
+  CHECK(!inner_)
+      << "Must cancel original input to FramedReader before setting a new one";
+  inner_ = std::move(inner);
+  inner_->onSubscribe(this->ref_from_this(this));
 }
 
 bool FramedReader::ensureOrAutodetectProtocolVersion() {
-  if (*protocolVersion_ != ProtocolVersion::Unknown) {
+  if (*version_ != ProtocolVersion::Unknown) {
     return true;
   }
 
   auto minBytesNeeded = std::max(
       FrameSerializerV0_1::kMinBytesNeededForAutodetection,
       FrameSerializerV1_0::kMinBytesNeededForAutodetection);
-  DCHECK(minBytesNeeded > 0);
+  DCHECK_GT(minBytesNeeded, 0);
   if (payloadQueue_.chainLength() < minBytesNeeded) {
     return false;
   }
 
-  DCHECK(minBytesNeeded > kFrameLengthFieldLengthV0_1);
-  DCHECK(minBytesNeeded > kFrameLengthFieldLengthV1_0);
+  DCHECK_GT(minBytesNeeded, kFrameLengthFieldLengthV0_1);
+  DCHECK_GT(minBytesNeeded, kFrameLengthFieldLengthV1_0);
 
-  bool recognized = FrameSerializerV1_0::detectProtocolVersion(
-                        *payloadQueue_.front(), kFrameLengthFieldLengthV1_0) !=
-      ProtocolVersion::Unknown;
-  if (recognized) {
-    *protocolVersion_ = FrameSerializerV1_0::Version;
+  auto const& firstFrame = *payloadQueue_.front();
+
+  auto detected = FrameSerializerV1_0::detectProtocolVersion(
+      firstFrame, kFrameLengthFieldLengthV1_0);
+  if (detected != ProtocolVersion::Unknown) {
+    *version_ = FrameSerializerV1_0::Version;
     return true;
   }
 
-  recognized = FrameSerializerV0_1::detectProtocolVersion(
-                   *payloadQueue_.front(), kFrameLengthFieldLengthV0_1) !=
-      ProtocolVersion::Unknown;
-  if (recognized) {
-    *protocolVersion_ = FrameSerializerV0_1::Version;
+  detected = FrameSerializerV0_1::detectProtocolVersion(
+      firstFrame, kFrameLengthFieldLengthV0_1);
+  if (detected != ProtocolVersion::Unknown) {
+    *version_ = FrameSerializerV0_1::Version;
     return true;
   }
 
-  error("could not detect protocol version from framing");
+  error("Could not detect protocol version from framing");
   return false;
 }
 
 void FramedReader::error(std::string errorMsg) {
   VLOG(1) << "error: " << errorMsg;
-  onError(std::runtime_error(std::move(errorMsg)));
+
+  if (DuplexConnection::Subscriber::subscription()) {
```

Comment:
so this is really the only change in this PR, is that correct? The rest is only renaming and reformatting? And unit tests.

Reply:
Should be yeah.  I'm looking to clean up some of the error/termination logic in `FramedReader`, and this was just the start of it.